src/java/arjdbc/postgresql/PostgreSQLRubyJdbcConnection.java

```diff
@@ -280,6 +286,56 @@ protected Connection newConnection() throws SQLException {
         return connection;
     }
 
+    // From what I can tell downCase isn't used for postgres but leaving it here to mimic the overridden method
+    @Override
+    protected ColumnData[] setupColumns(
+            final ThreadContext context,
+            final Connection connection,
+            final ResultSetMetaData resultMetaData,
+            final boolean downCase) throws SQLException {
+
+        final int columnCount = resultMetaData.getColumnCount();
+        final ColumnData[] columns = new ColumnData[columnCount];
+        final IRubyObject adapter = getAdapter();
+        final PgResultSetMetaDataWrapper mdWrapper = new PgResultSetMetaDataWrapper((PgResultSetMetaData) resultMetaData);
+
+        for ( int i = 1; i <= columnCount; i++ ) { // metadata is one-based
+            String name = resultMetaData.getColumnLabel(i);
+            if ( downCase ) {
+                name = name.toLowerCase();
+            } else {
+                name = caseConvertIdentifierForRails(connection, name);
+            }
+
+            final int columnType = resultMetaData.getColumnType(i);
+            final Field field = mdWrapper.getField(i);
+            final IRubyObject[] args = { context.runtime.newFixnum(field.getOID()), context.runtime.newFixnum(field.getMod()), cachedString(context, name) };
+            columns[i - 1] = new ColumnData(name, columnType, i, adapter.callMethod(context, "get_oid_type", args));
+        }
+
+        return columns;
+    }
+
+    // This should probably apply to all adapters but didn't want to affect them for now
+    protected static IRubyObject newResult(final ThreadContext context, ColumnData[] columns, IRubyObject rows) {
+        final RubyClass Result = getResult(context.runtime);
+        return Result.newInstance(context, columnsToArray(context, columns), rows, columnTypes(context, columns), Block.NULL_BLOCK); // Result.new
+    }
+
+    private static RubyHash columnTypes(ThreadContext context, ColumnData[] columns) {
+        RubyHash types = RubyHash.newHash(context.runtime);
+
+        for ( int i = 0; i < columns.length; i++ ) {
+            IRubyObject type = columns[i].getRubyType();
```

Comment:
@enebo This is where the `rubyType` you asked about is used

Reply:
This ultimately gets sent to the `AR::Result` object which is what uses the type object